packages/gestalt-datepicker/src/DatePicker.js

```diff
@@ -131,6 +131,10 @@ type Props = {|
    * Required for date range selection. Pass a ref object to DatePicker to autofocus on the unselected date range field. See the [date range picker example](https://gestalt.pinterest.systems/web/datepicker#rangePicker) to learn more.
    */
   ref?: Element<'input'>, // eslint-disable-line react/no-unused-prop-types
+  /**
+   * Show a select list for quick selection of year and month. See the [showMonthYearDropdown example](https://gestalt.pinterest.systems/web/datepicker#showMonthYearDropdown) to learn more.
+   */
+  showMonthYearDropdown?: boolean,
```

Comment:
Do we have a use case for only showing the month dropdown, or only showing the year? I wonder if this would be better as something like 
```js
dropdowns?: $ReadOnlyArray<'month' | 'year'>,

...

<ReactDatePicker
  dropdownMode={dropdowns.length > 0 ? 'select' : undefined}
  showMonthDropdown={dropdowns.includes('month')}
  showYearDropdown={dropdowns.includes('year')}
```